pkg/controller/reconciler/nameserverdelegator_reconciler.go

```diff
@@ -0,0 +1,94 @@
+package reconciler
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/mishudark/errors"
+	"github.com/oslokommune/okctl/pkg/client"
+	"github.com/oslokommune/okctl/pkg/controller/resourcetree"
+	"github.com/oslokommune/okctl/pkg/domain"
+)
+
+const nsRecordValidationIntervalSeconds = 15
+
+// NameserverHandlerReconcilerResourceState contains data extracted from the desired state
+type NameserverHandlerReconcilerResourceState struct {
+	PrimaryHostedZoneFQDN string
+	Nameservers           []string
+}
+
+// nameserverDelegationReconciler contains service and metadata for the relevant resource
+type nameserverDelegationReconciler struct {
+	commonMetadata *resourcetree.CommonMetadata
+
+	client        client.NameserverRecordDelegationService
+	domainService client.DomainService
+}
+
+// SetCommonMetadata saves common metadata for use in Reconcile()
+func (z *nameserverDelegationReconciler) SetCommonMetadata(metadata *resourcetree.CommonMetadata) {
+	z.commonMetadata = metadata
+}
+
+// Reconcile knows how to do what is necessary to ensure the desired state is achieved
+func (z *nameserverDelegationReconciler) Reconcile(node *resourcetree.ResourceNode) (*ReconcilationResult, error) {
+	resourceState, ok := node.ResourceState.(NameserverHandlerReconcilerResourceState)
+	if !ok {
+		return nil, errors.New("error casting nameserverhandler state")
+	}
+
+	switch node.State {
+	case resourcetree.ResourceNodeStatePresent:
+		err := z.commonMetadata.Spin.Start("Nameserver delegation")
```

Comment:
I interpret reconcilers to be a pure back-end thing, something that knows how to adjust state. A spinner is a user interface thing, thus I don't think it should belong here. Is it possible to put this spinner loggic somewhere else? 

Or is it hard? I'm seeing lots `spinner.Start` in `pkg/client/core/service_*.go`.

Reply:
Good point. I actually moved it there from the service_* file due to things breaking. I'll look into it and see if I can make it work in the original place

Reply:
Actually, unless you or @ivaruf has any good ideas about where to place the spinner functionality, I'll have to ask @paulbes. The spinner gets used by both the nameserver delegator and the domainService. I either have to create subspinners somewhere or do it like this. 

I'm actually concidering if maybe all the spinner functionality should be moved into the reconcilers. It makes sense in a operation scope kind of way. Thats where the operation starts and stops.

Reply:
OK, after discussion we can't see a good solution for now. Discuss with @paulbes later. Leave as is for now.